coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/services/PartsTreeQueryService.java

```
@@ -30,56 +38,55 @@
-
-        final Optional<PartInfo> vehicle = partsTree.getPartInfos().stream()
-                .filter(p -> "vehicle".equals(p.getPartTypeName()))
-                .findFirst();
-        if (vehicle.isEmpty()) {
-            return Optional.empty();
-        }
-
-        final String vehicleObjectId = vehicle.get().getPart().getObjectIDManufacturer();
-        if (!Objects.equals(vehicleObjectId, request.getVin())) {
-            return Optional.empty();
-        }
-
-        return Optional.of(partsTree);
-    }
+    private final PartAspectRepository aspectRepository;
+    /**
+     * Repository for retrieving {@link PartAttributeEntity} data.
+     */
+    private final PartAttributeRepository attributeRepository;
+    /**
+     * Mapper from entities to {@link PartRelationshipsWithInfos} DTO.
+     */
+    private final PartRelationshipEntityListToDtoMapper mapper;
+    /**
+     * PRS configuration settings.
+     */
+    private final PrsConfiguration configuration;
 
     /**
-     * Get a PartsTree for a part
+     * Get a parts tree for a {@link PartsTreeByObjectIdRequest}.
      *
      * @param request Request.
      * @return PartsTree with parts info.
      */
     public Optional<PartRelationshipsWithInfos> getPartsTree(final PartsTreeByObjectIdRequest request) {
-        final PartId partId = PartId.builder()
-                .withOneIDManufacturer(request.getOneIDManufacturer())
-                .withObjectIDManufacturer(request.getObjectIDManufacturer())
-                .build();
-
-        final PartRelationshipsWithInfos partsTree = stubResourcesHelper.getStubbedPartsTreeData();
-
-        final boolean isPresent = partsTree.getPartInfos().stream()
-                .anyMatch(p -> p.getPart().equals(partId));
-        if (!isPresent) {
+        final int maxDepth = Integer.min(request.getDepth().orElse(Integer.MAX_VALUE), configuration.getPartsTreeMaxDepth());
```

A general observation is this will silently hide the fact that the api caller depth value is reduced to what we have in out configuration. How about if we throw an error and indicate to caller that we support depth > 100 for example.
And we should do this validation a bit earlier on the request object to save some db calls.

---

Great point; implemented